src/Aspire.Hosting/Dashboard/DashboardLifecycleHook.cs

```diff
@@ -154,4 +189,111 @@ private void ConfigureAspireDashboardResource(IResource dashboardResource)
             }
         }));
     }
+
+    private async Task WatchDashboardLogsAsync()
+    {
+        async Task<string?> GetDashboardResourceIdentifier()
+        {
+            string? dashboardResourceId = null;
+
+            try
+            {
+                // The dashboard resource isn't immediately available so watch for it.
+                // Wait for dashboard to startup and be reported before giving up.
+                using var cts = new CancellationTokenSource(TimeSpan.FromSeconds(30));
+
+                await foreach (var notification in resourceNotificationService.WatchAsync(cts.Token))
+                {
+                    // We want to capture the replica of the dashboard resource. This will allow us to get the logs for the dashboard.
+                    // When the resource id is not the same as the resource name, it's the replica.
+                    if (notification.Resource.Name == KnownResourceNames.AspireDashboard && notification.ResourceId != notification.Resource.Name)
+                    {
+                        dashboardResourceId = notification.ResourceId;
+                        break;
+                    }
+                }
+            }
+            catch (OperationCanceledException)
+            {
+            }
+
+            return dashboardResourceId;
+        }
+
+        if (await GetDashboardResourceIdentifier().ConfigureAwait(false) is not string dashboardResourceId)
+        {
+            // The only way this can happen is if DCP is unhealthy, so log a warning.
+            distributedApplicationLogger.LogWarning("Failed to get the dashboard logs.");
+            return;
+        }
+
+        var loggerCache = new Dictionary<string, ILogger>();
```

Comment:
Assuming zero trust between hosting and dashboard, should there be a limit of cached loggers?

Reply:
Hmm, I was going to say that you could consider removing altogether since there is already a cache inside LoggerFactory: https://github.com/dotnet/runtime/blob/f0991722db63e0b4962439469efb4ea1b72923f3/src/libraries/Microsoft.Extensions.Logging/src/LoggerFactory.cs#L139-L162

But it doesn't limit the number of loggers either...

Reply:
Yea, not adding a limit here.